samples/nrf9160/ciphersuites/README.rst

```diff
@@ -0,0 +1,126 @@
+This certificate is provided in the :file:`samples/nrf9160/https_client/cert` folder.
+
+To connect to other servers, you might need to provision a different certificate.
+See :ref:`cert_dwload` for more information.
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/nrf9160/ciphersuites`
+
+.. include:: /includes/build_and_run_nrf9160.txt
+
+Testing
+=======
+
+|test_sample|
+
+1. |connect_kit|
+#. |connect_terminal|
+#. Observe that the sample starts, provisions certificates, and connects to the LTE network.
+#. Observe that the sample iterates through a list of ciphersuites, attempting a connection to ``example.com`` with each one of them, showing either a successful or an unsuccessful connection.
+
+Sample output
+=============
+
+The sample shows the following output:
+
+.. code-block:: console
+
+   ciphersuites sample started
+   certificate match
+   waiting for network.. OK
+   trying all ciphersuites to find which ones are supported...
+   trying ciphersuite: TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384
+   connecting to example.com... connect() failed, err: 95, Operation not supported on socket
+   trying ciphersuite: TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA
+   connecting to example.com... connect() failed, err: 95, Operation not supported on socket
+   trying ciphersuite: TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256
+   connecting to example.com... connect() failed, err: 95, Operation not supported on socket
+   trying ciphersuite: TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA
+   connecting to example.com... connect() failed, err: 95, Operation not supported on socket
+   trying ciphersuite: TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
+   connecting to example.com... Connected.
+   trying ciphersuite: TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
+   connecting to example.com... Connected.
+   trying ciphersuite: TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
+   connecting to example.com... Connected.
+   trying ciphersuite: TLS_PSK_WITH_AES_256_CBC_SHA
+   connecting to example.com... connect() failed, err: 95, Operation not supported on socket
+   trying ciphersuite: TLS_PSK_WITH_AES_128_CBC_SHA256
+   connecting to example.com... connect() failed, err: 95, Operation not supported on socket
+   trying ciphersuite: TLS_PSK_WITH_AES_128_CBC_SHA
+   connecting to example.com... connect() failed, err: 95, Operation not supported on socket
+   trying ciphersuite: TLS_PSK_WITH_AES_128_CCM_8
+   connecting to example.com... connect() failed, err: 95, Operation not supported on socket
+   trying ciphersuite: TLS_EMPTY_RENEGOTIATIONINFO_SCSV
+   connecting to example.com... connect() failed, err: 95, Operation not supported on socket
+
+   Ciphersuite support summary for host `example.com`:
+   TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384: No
+   TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA: No
+   TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256: No
+   TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA: No
+   TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA: Supported
```

Comment:
Would it make more sense to have `Yes` / `No` or `Supported` / `Not supported` instead of `Supported` / `No`?

Reply:
@eivindj-nordic I thought about that while working on it, and the reason for the current implementation is that it's more immediate to see which ones are supported. For example, `supported` and `not supported` look too similar and since the strings are not aligned in a tabular fashion it would be less immediate. Same with `no` and `yes`.